cas-server-core/src/main/java/org/jasig/cas/authentication/principal/CachingPrincipalAttributesRepository.java

```diff
@@ -332,4 +352,44 @@ protected void finalize() throws Throwable {
         close();
         super.finalize();
     }
+
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .append("attributeRepository", attributeRepository)
+                .append("cache", cache)
+                .append("cacheName", cacheName)
+                .append("durationTimeUnit", duration.getTimeUnit())
+                .append("durationAmount", duration.getDurationAmount())
+                .append("mergingStrategy", mergingStrategy)
+                .toString();
+    }
+
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (obj == this) {
+            return true;
+        }
+        if (obj.getClass() != getClass()) {
+            return false;
+        }
+        final CachingPrincipalAttributesRepository rhs = (CachingPrincipalAttributesRepository) obj;
+        final EqualsBuilder builder = new EqualsBuilder();
+        return builder
```

Comment:
Because the cache name is tied to each instance, and is dynamically constructed with UUID every time the object is spin up.